cmake/CMakeLists.txt

```
@@ -0,0 +1,160 @@
+project(usb-1.0)
+cmake_minimum_required(VERSION 3.10)
+set(CMAKE_C_USE_RESPONSE_FILE_FOR_INCLUDES   OFF)
+set(CMAKE_CXX_USE_RESPONSE_FILE_FOR_INCLUDES OFF)
+
+include(CheckIncludeFiles)
+include(CheckFunctionExists)
+include(CheckSymbolExists)
+include(CheckStructHasMember)
+include(CheckCCompilerFlag)
+include(CTest)
+
+check_function_exists(clock_gettime             HAVE_CLOCK_GETTIME)
+check_function_exists(pthread_condattr_setclock HAVE_PTHREAD_CONDATTR_SETCLOCK)
+check_function_exists(pthread_setname_np        HAVE_PTHREAD_SETNAME_NP)
+check_function_exists(pthread_threadid_np       HAVE_PTHREAD_THREADID_NP)
+check_function_exists(eventfd                   HAVE_EVENTFD)
+check_function_exists(pipe2                     HAVE_PIPE2)
+check_function_exists(syslog                    HAVE_SYSLOG)
+
+check_include_files(asm/types.h      HAVE_ASM_TYPES_H) 
+check_include_files(sys/eventfd.h    HAVE_EVENTFD) 
+check_include_files(dlfcn.h          HAVE_DLFCN_H)
+check_include_files(inttypes.h       HAVE_INTTYPES_H)
+check_include_files(stdint.h         HAVE_STDINT_H)
+check_include_files(stdio.h          HAVE_STDIO_H)
+check_include_files(stdlib.h         HAVE_STDLIB_H)
+check_include_files(string.h         HAVE_STRING_H)
+check_include_files(strings.h        HAVE_STRINGS_H)
+check_include_files(sys/stat.h       HAVE_SYS_STAT_H)
+check_include_files(sys/time.h       HAVE_SYS_TIME_H)
+check_include_files(sys/types.h      HAVE_SYS_TYPES_H)
+check_include_files(unistd.h         HAVE_UNISTD_H)
+check_include_files("stdlib.h;stdarg.h;string.h;float.h" STDC_HEADERS)
+
+check_symbol_exists(EFD_CLOEXEC     "sys/eventfd.h" HAVE_DECL_EFD_CLOEXEC)
+check_symbol_exists(EFD_NONBLOCK    "sys/eventfd.h" HAVE_DECL_EFD_NONBLOCK)
+check_symbol_exists(TFD_CLOEXEC     "sys/eventfd.h" HAVE_DECL_TFD_CLOEXEC)
+check_symbol_exists(timerfd_create  "sys/timerfd.h" HAVE_TIMERFD) 
+
+check_struct_has_member("struct timespec" tv_sec time.h HAVE_STRUCT_TIMESPEC)
+
+if (BUILD_SHARED_LIBS)
+  set(libusb_BUILD_SHARED_LIBS_DEFAULT ON)
+else (BUILD_SHARED_LIBS)
+  set(libusb_BUILD_SHARED_LIBS_DEFAULT OFF)
+endif (BUILD_SHARED_LIBS)
+
+option(libusb_BUILD_SHARED_LIBS "Build Shared Libraries for libusb" ${libusb_BUILD_SHARED_LIBS_DEFAULT})
+option(libusb_BUILD_TESTING "Build Tests" ON)
+option(ENABLE_LOGGING "Enable Logging" ON)
+
+if (WIN32)
+    set(PLATFORM_WINDOWS 1 CACHE INTERNAL "")
+endif()
+if (UNIX)
+    set(PLATFORM_POSIX 1  CACHE INTERNAL "")
+endif()
```

Pardon my pedantic eye:
```suggestion
if(WIN32)
    set(PLATFORM_WINDOWS 1 CACHE INTERNAL "")
endif()
if(UNIX)
    set(PLATFORM_POSIX 1  CACHE INTERNAL "")
endif()
```
In CMake, and `if` is not an operator as in C-like languages, but more like a function, and it is a common practice _not to use_ space between an `if` and `(`. It alsom makes it consisntent with `endif()`, where you'd rarely see a space like `endif ()`.

---
Same for all other places.

---

Done. Thanks